pkg/operator/utils/gcp/utils_test.go

```diff
@@ -171,39 +182,20 @@ func TestIgnoreInvalidProjectPermissions(t *testing.T) {
 	tests := []struct {
 		name             string
 		origPermList     []string
+		allowed          bool
 		setupGCPResponse func(*mock.MockClient)
 	}{
 		{
-			name: "receive invalid permission and try again",
+			name: "one invalid permission",
 			origPermList: []string{
 				"allowedPerm",
 				"invalidPerm",
 			},
 			setupGCPResponse: func(mockClient *mock.MockClient) {
-				// first reject the "invalidPerm"
-				firstMock := mockClient.EXPECT().TestIamPermissions(projectName, gomock.Any()).DoAndReturn(
-					func(projectName string, permRequest *cloudresourcemanager.TestIamPermissionsRequest) (*cloudresourcemanager.TestIamPermissionsResponse, error) {
-
-						if len(permRequest.Permissions) != 2 {
-							return nil, fmt.Errorf("expected length 2 list of permissions to test")
-						}
-
-						return nil, &googleapi.Error{
-							Code:    400,
-							Message: "Permission invalidPerm is not valid for this resource.",
-						}
-					})
-
-				// now allow the perm list w/o invalidPerm
-				secondMock := mockClient.EXPECT().TestIamPermissions(projectName, gomock.Any()).DoAndReturn(
+				// allow the perm list w/o invalidPerm
+				mockCall := mockClient.EXPECT().TestIamPermissions(projectName, gomock.Any()).Times(1).DoAndReturn(
```

Comment:
There is only one call to `TestIamPermissions` inside `CheckPermissionsAgainstPermissionList`. It takes all permissions and returns valid permissions. I do not think other mocks are required here.

Reply:
Is this being missed because of a missing `defer Finish()` on the generated mockcontroller down by line 255?

Reply:
I am not really sure why we are expecting 2 calls to `TestIamPermissions` here. From what I understand (from the code snippet below of `CheckPermissionsAgainstPermissionList` function), we should have 1 call to `TestIamPermissions` for 100 permissions (i.e. chunksize) and since we have only 2 permissions there should be only 1 call. 

```
	chunkSize := 100
	permLen := len(filteredPermList)
	for i := 0; i < permLen; i += chunkSize {
		end := i + chunkSize
		if end > permLen {
			end = permLen
		}
		req := &cloudresourcemanager.TestIamPermissionsRequest{Permissions: filteredPermList[i:end]}
		resp, err := gcpClient.TestIamPermissions(projectName, req)
		if err != nil {
			return false, fmt.Errorf("error testing permissions: %v", err)
		}
		allowedPerms.Insert(resp.Permissions...)
	}
```

Correct me if I am missing something here :(

Reply:
Maybe I'm missing something in this discussion. I only see one EXPECT() for TestIamPermissions()...

Reply:
Ok, we got this figured out after talking.